R/dump.R

```diff
@@ -0,0 +1,290 @@
+#' Query a database for all information relevant to a particular touchstone
+#' (including version), and write to CSV files, which would then be fit for
+#' stoner to use as its input to re-populate the database with the same
+#' touchstone data. This would be useful in a workflow where you wanted to
+#' clone a touchstone, alter a few things, and upload as a new version.
+#'
+#' Note that this is not "lossless"; only things that Stoner has the capability
+#' of adding, will be stored in the CSVs. For example (at present), Stoner
+#' does nothing with coverage sets, so while the scenario_descriptions will
+#' be saved (since they can be re-added),
+#' scenario_description.focal_coverage_set will not be dumped, because
+#' Stoner cannot put it back. Coverage sets will need to be wired up
+#' separately after a Stoner import, as before.
+
+#'
+#' @title Dump touchstone from database to CSV.
+#' @export
+#' @param con Active connection to the DB which the data will be dumped from.
+#' @param touchstone the touchstone, including version
+#' @param path the path where output CSVs should be created.
+#' @param include_deps also dump db dependencies that Stoner can't
+#' currently add - db_disease.csv, db_modelling_group.csv,
+#' db_demographic_statistic_type.csv and db_demographic_source.csv.
+
+stone_dump <- function(con, touchstone, path, include_deps = FALSE) {
```

Comment:
The design here, with a lot of functions within a function, makes me wonder if what you really want is some sort of OOP approach to this?

Reply:
I was only really thinking code layout - having some smallish modular functions for each CSV file that gets dumped (more or less), with them all using the 4 args sent to stone_dump as a whole.

Reply:
I only really had layout in mind - smallish functions, one per CSV file - all needing the same 4 args that the surrounding stone_dump has... essentially, trying to stop it being a long sequential mess...!